signer/src/ecdsa.rs

```diff
@@ -0,0 +1,183 @@
+//! # ECDSA Signing and Verification
```

Comment:
Did you write this manually? How do we feel about using AI for documentation boilerplate?

Reply:
I co-authored this with ChatGPT, using my normal AI-collaboration-workflow. 1. Ask the AI to do something for me, 2. Rewrite most of it because the AI doesn't write what I want it to 😅

Reply:
What is `//!`? I'm not familiar with it, and couldn't find anything with a quick google search. Is it for "top-level" module comments?

Reply:
> What is `//!`? I'm not familiar with it, and couldn't find anything with a quick google search. Is it for "top-level" module comments?

Yeah, here it's a module [doc comment](https://doc.rust-lang.org/reference/comments.html#doc-comments). In general it's a special kind of doc comment that applies to what comes before the comment (rather than what comes after).

Reply:
Thanks for chiming in @djordon :pray: